### PR TITLE
Added Documentation for Azure Remote

### LINKED ISSUE #5151: Add docs for missing Azure remote options
For azure remote:

```python
               "exclude_environment_credential": Bool,
               "exclude_visual_studio_code_credential": Bool,
               "exclude_shared_token_cache_credential": Bool,
               "exclude_managed_identity_credential": Bool,
```

are missing from docs, but critical in some environments. Docs would have saved me a lot of time. At least this ticket hopefully can also help other folks who need to make DVC avoid certain credentials.

### PR DESCRIPTION
**Pull request by aayushk9**
Closes #5151 
Integrated documentation for azure remote in `user-guide/data-management/remote-storage/azure-blob-storage `.

### CODE REVIEW

**Review by shcheklein — CHANGES_REQUESTED** [review]
Hey @aayushk9 ! thanks. there is a dedicated page for the Azure storage - https://dvc.org/doc/user-guide/data-management/remote-storage/azure-blob-storage . Let's please use that one.

**Comment by aayushk9** [comment]
@shcheklein I've made the changes. Could you please review them?

File: content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md [hunk 1]
```diff
@@ -195,6 +195,26 @@ from an [Azure CLI configuration file] (typically managed with [az config]):
   https://learn.microsoft.com/en-us/cli/azure/azure-cli-configuration#cli-configuration-file
 [az config]: https://docs.microsoft.com/en-us/cli/azure/config
 
+## Configure Azure Remote with environment variables
```
Comment (shcheklein) [inline content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md:198]: This should go to the `More configuration parameters` section.

Let's please move it there. Please keep the same style as we have for other params.

**Comment by shcheklein** [comment]
@aayushk9 thanks, I've put a comment to do the next step here!

**Comment by aayushk9** [comment]
@shcheklein Thank you for your feedback. I've made the changes, please let me know if there are any further edits needed.

File: content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md (hunk 1 above)
Comment (aayushk9) [reply content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md]: I've removed sample configuration as well, is that okay?

File: content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md [hunk 2]
```diff
@@ -219,3 +219,14 @@ See `dvc remote modify` for more command usage details.
 
 [Azure documentation]:
   https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/storage/azure-storage-blob#other-client--per-operation-configuration
+
+### Azure Remote Configuration
```
Comment (shcheklein) [inline content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md:223]: I don't think we need a separate `###` - it should be just part of the existing list (`url`, timeouts, ect). it is the same: some additional config option.

File: content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md [hunk 3]
```diff
@@ -219,3 +219,14 @@ See `dvc remote modify` for more command usage details.
 
 [Azure documentation]:
   https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/storage/azure-storage-blob#other-client--per-operation-configuration
+
+### Azure Remote Configuration 
+
+- `"exclude_environment_credential"`: Bool. If `true`, excludes the environment credential source for Azure Remote.
```
Comment (shcheklein) [inline content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md:225]: please keep the style exactly the same as with the list above: 

`:` -> `-`

start the description from the low case. no need to specify type.

**Comment by aayushk9** [comment]
@shcheklein Have a look :)

File: content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md (hunk 3 above)
Comment (aayushk9) [reply content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md]: removed if statements and kept `bool` , is that fine?

File: content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md [hunk 4]
```diff
@@ -219,3 +219,11 @@ See `dvc remote modify` for more command usage details.
 
 [Azure documentation]:
   https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/storage/azure-storage-blob#other-client--per-operation-configuration
+
+- `exclude_environment_credential` - bool
```
Comment (shcheklein) [inline content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md:223]: hmm, there are not descriptions now?

**Comment by aayushk9** [comment]
@shcheklein I've added the missing descriptions.

File: content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md [hunk 5]
```diff
@@ -219,3 +219,11 @@ See `dvc remote modify` for more command usage details.
 
 [Azure documentation]:
   https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/storage/azure-storage-blob#other-client--per-operation-configuration
+
+- `exclude_environment_credential` - bool. If `true`, excludes the environment credential source for Azure Remote.
```
Comment (shcheklein) [inline content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md:223]: let's move it before the `Azure documentation` please (it should be a single list).

File: content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md [hunk 6]
```diff
@@ -219,3 +219,11 @@ See `dvc remote modify` for more command usage details.
...
+
+- `exclude_visual_studio_code_credential` - bool. If `true`, excludes Visual Studio Code credential source for Azure Remote.
```
Comment (shcheklein) [inline content/docs/user-guide/data-management/remote-storage/azure-blob-storage.md:225]: `bool` is not needed (see how we do other items - e.g. we don't specify `int` for the connection_timeout`.

**Review by shcheklein — CHANGES_REQUESTED** [review]
hey @aayushk9 , sorry for the delay. I left some comments. We are getting there. thanks.

**Comment by aayushk9** [comment]
Hey @shcheklein, thanks for pointing out those changes. I have included the requested modifications and please let me know if any further changes are needed.

**Comment by shcheklein** [comment]
thanks @aayushk9 , I've fixed it a bit and it's ready to merged!

**Review by shcheklein — APPROVED** [review]

**Pull request merged**